app/models/application_plan.rb

```diff
@@ -16,10 +16,14 @@ class ApplicationPlan < Plan
     if provider == :all || provider.blank?
       {}
     else
-      where(services: { account_id: provider.id }).joins(:service).references(:service).readonly(false)
+      by_provider_id(provider).readonly(false)
     end
   }
 
+  scope :by_provider_id, ->(provider_id) { joins(:service).references(:service).where.has { service.account_id == provider_id } }
+
+  scope :enterprise, -> { where.has { (system_name =~ '%enterprise%') } }
```

Comment:
I'd avoid `enterprise` here. We wanted to remove it from the plan_rules when opensourcing but did not succeed yet. It would be good to have some way to define which plans (it can only be one now) are going to be `protected` from deletion.

Reply:
Ok, but one question about this:

> It would be good to have some way to define which plans (it can only be one now)

What do you mean that it can be only one now? 😱

Reply:
Btw I am not sure if out of urgency we should add a comment now to don't use this scope but merge it anyway and in another PR doing that refactoring that you say, because right now we have active in SaaS the suspend/delete always without these exceptions 😞

Reply:
It already appears and there is logic with it in ProviderUpgrade 😝